src/viam/app/app_client.py

```diff
@@ -0,0 +1,791 @@
+
+    async def location_auth(self):
+        raise NotImplementedError()
+
+    async def create_location_secret(self):
+        raise NotImplementedError()
+
+    async def delete_location_secret(self):
+        raise NotImplementedError()
+
+    async def get_robot(self, robot_id: str) -> Robot:
+        """Get a robot.
+
+        Args:
+            id (str): ID of the robot to get.
+
+        Raises:
+            GRPCError: If an invalid robot ID is passed.
+
+        Returns:
+            viam.proto.app.Robot: The robot.
+        """
+        request = GetRobotRequest(id=robot_id)
+        response: GetRobotResponse = await self._location_client.GetRobot(request, metadata=self._metadata)
+        return response.robot
+
+    async def get_rover_rental_parts(self):
+        raise NotImplementedError()
+
+    async def get_robot_parts(self, robot_id: str) -> List[RobotPart]:
+        """Get a list of all the parts under a specific robot.
+
+        Args:
+            robot_id (str): ID of the robot to get parts from.
+
+        Raises:
+            GRPCError: If an invalid robot ID is passed.
+
+        Returns:
+            List[viam.app.app_client.RobotPart]: The list of robot parts.
+        """
+        request = GetRobotPartsRequest(robot_id=robot_id)
+        response: GetRobotPartsResponse = await self._location_client.GetRobotParts(request, metadata=self._metadata)
+        return [RobotPart.from_proto(robot_part=part) for part in response.parts]
+
+    async def get_robot_part(self, robot_part_id: Optional[str], dest: Optional[str] = None, indent: Optional[int] = None) -> RobotPart:
+        """Get a robot part.
+
+        Args:
+            robot_part_id (str): ID of the robot part to get.
+            dest (Optional[str]): Optional filepath to write the robot part's config file in JSON format to.
+            indent (Optional[int]): Optional size (in number of spaces) of indent when writing config to `dest`. Defaults to 4.
+
+        Raises:
+            GRPCError: If an invalid robot part ID is passed.
+
+        Returns:
+            viam.app.app_client.RobotPart: The robot part.
+        """
+        indent = indent if indent else 4
+        request = GetRobotPartRequest(id=robot_part_id)
+        response: GetRobotPartResponse = await self._location_client.GetRobotPart(request, metadata=self._metadata)
+
+        if dest:
+            try:
+                file = open(dest, "w")
+                file.write(f"{json.dumps(json.loads(response.config_json), indent=indent)}")
+            except Exception as e:
+                LOGGER.err(f"Failed to write config JSON to file {dest}", exc_info=e)
+
+        return RobotPart.from_proto(robot_part=response.part)
+
+    async def get_robot_part_logs(
+        self,
+        robot_part_id: str,
+        filter: Optional[str] = None,
+        dest: Optional[str] = None,
+        errors_only: Optional[bool] = None,
```

Comment:
Docstring says this defaults to `True`. So have it default to `True` in the signature.